vaultclient/createAccount.go

```diff
@@ -78,67 +79,73 @@ func (s *CreateAccountInput) SetExternalAccountID(v string) *CreateAccountInput
+	}
+	if s.ExternalAccountID != nil {
+		formData.Set("externalAccountId", *s.ExternalAccountID)
+	}
+	return formData
+}
+
 // CreateAccount API operation creates a new Vault account
 // and adds the ability to pass a context and additional request options.
 //
 // The context must be non-nil and will be used for request cancellation. If
 // the context is nil a panic will occur. In the future the SDK may create
 // sub-contexts for http.Requests. See https://golang.org/pkg/context/
 // for more information on using Contexts.
-func (c *Vault) CreateAccount(ctx aws.Context, input *CreateAccountInput, opts ...request.Option) (*CreateAccountOutput, error) {
-	req, out := c.CreateAccountRequest(input)
-	req.SetContext(ctx)
-	req.ApplyOptions(opts...)
-	return out, req.Send()
-}
+func (c *Vault) CreateAccount(ctx context.Context, input *CreateAccountInput) (*CreateAccountOutput, error) {
+	if err := input.Validate(); err != nil {
+		return nil, err
+	}
 
-// CreateAccountRequest generates a "aws/request.Request" representing the
-// client's request for the CreateAccount operation. The "output" return
-// value will be populated with the request's response once the request completes
-// successfully.
-//
-// Use "Send" method on the returned Request to send the API call to the service.
-// the "output" return value is not valid until after Send returns without error.
-func (c *Vault) CreateAccountRequest(input *CreateAccountInput) (req *request.Request, output *CreateAccountOutput) {
-	op := &request.Operation{
-		Name:       opCreateAccount,
-		HTTPMethod: "POST",
-		HTTPPath:   "/",
+	resp, err := c.makeAWSRequest(ctx, opCreateAccount, input.getUrlValues())
+	if err != nil {
+		return nil, err
+	}
+
+	respBody, err := c.handleAWSResponse(resp)
+	if err != nil {
+		return nil, err
 	}
 
-	if input == nil {
-		input = &CreateAccountInput{}
+	var output CreateAccountOutput
+	if err := json.Unmarshal(respBody, &output); err != nil {
+		return nil, fmt.Errorf("failed to parse response: %w", err)
 	}
 
-	output = &CreateAccountOutput{}
-	req = c.newRequest(op, input, output)
-	return
+	return &output, nil
 }
 
 // Account contains information about a Vault account.
 type Account struct {
-	AccountData *AccountData `type:"structure" locationName:"data"`
+	AccountData *AccountData `type:"structure" locationName:"data" json:"data"`
 }
 
 type AccountData struct {
-	Arn         *string    `locationName:"arn"`
-	Name        *string    `locationName:"name"`
-	Email       *string    `locationName:"emailAddress"`
-	ID          *string    `locationName:"id"`
-	QuotaMax    *int64     `locationName:"quotaMax"`
-	CreateDate  *time.Time `locationName:"createDate"`
-	CanonicalID *string    `locationName:"canonicalId"`
-	AliasList   []*string  `locationName:"aliasList"`
+	Arn         *string    `locationName:"arn" json:"arn"`
+	Name        *string    `locationName:"name" json:"name"`
+	Email       *string    `locationName:"emailAddress" json:"emailAddress"`
```

Comment:
nit: json tag not _required_ for the simpler cases (arm, name, id), as this is the same as the default (but does not harm either)